js/controllers/grid/queries/ReadQueryHandler.js

```diff
@@ -130,8 +134,15 @@
 			showFetchedParticipantsList_ = function(ajaxContext, jsonData) {
 
 		var processedJsonData = this.handleJson(jsonData),
-				$participantsListContainer = $(
-				'#participantsListPlaceholder', this.getHtmlElement());
+				$participantsListContainer = $('#participantsListPlaceholder', this.getHtmlElement()),
+				$leaveQueryButton = $('.leaveQueryForm', this.getHtmlElement());
+
+		if (processedJsonData.showLeaveQueryButton == true && $leaveQueryButton.hasClass('pkp_helpers_display_none')) {
+			$leaveQueryButton.removeClass('pkp_helpers_display_none');
+		}
+		if (processedJsonData.showLeaveQueryButton == false && !$leaveQueryButton.hasClass('pkp_helpers_display_none')) {
+			$leaveQueryButton.addClass('pkp_helpers_display_none');
+		}
```

Comment:
It might be a little clearer here to just use the jQuery methods `.show()` and `.hide()`:

```
if (processedJsonData.showLeaveQueryButton) {
  $leaveQueryButton.show();
} else {
  $leaveQueryButton.hide();
}
```

Reply:
check